src/components/history/HistoryFilters.tsx

```diff
@@ -1,26 +1,83 @@
 import { useState } from "react";
 import { OutputGranuleExtentFlagOptions, OutputSamplingGridType, RasterResolution, Adjust, FilterParameters, FilterAction } from "../../types/historyPageTypes";
 import DatePicker from "react-datepicker";
 import "react-datepicker/dist/react-datepicker.css";
 import { Formik } from "formik";
 
+export const getFilterParameters = (currentFilters: FilterParameters, limit?: number, after?: string): UserProductQueryVariables => {
+    const productVariablesObject: ProductQueryParameters = {}
+    const {cycle, pass, scene, outputGranuleExtentFlag, outputSamplingGridType, startDate, endDate} = currentFilters
+
+    if(typeof limit !== 'undefined') productVariablesObject.limit = limit
+    if(typeof after !== 'undefined') productVariablesObject.after = after
+    if(cycle !== 'none') {
+        productVariablesObject.cycle = parseInt(cycle)
+    }
+    if (pass !== 'none') {
+        productVariablesObject.pass = parseInt(pass)
+    }
+    if (scene !== 'none') {
+        productVariablesObject.scene = parseInt(scene)
+    }
+    if (outputGranuleExtentFlag.length > 0) {
+        let outputGranuleExtentFlagToUse = null
+        if(outputGranuleExtentFlag.includes('128 x 128')) outputGranuleExtentFlagToUse = false
+        if(outputGranuleExtentFlag.includes('256 x 128')) outputGranuleExtentFlagToUse = true
+        if(outputGranuleExtentFlagToUse !== null) productVariablesObject.outputGranuleExtentFlag = outputGranuleExtentFlagToUse
+    }
+    if (outputSamplingGridType.length > 0) {
+        let outputSamplingGridTypeToUse = null
+        if(outputSamplingGridType.includes('UTM')) outputSamplingGridTypeToUse = 'UTM'
+        if(outputSamplingGridType.includes('LAT/LON')) outputSamplingGridTypeToUse = 'GEO'
+        if(outputSamplingGridTypeToUse !== null) productVariablesObject.outputSamplingGridType = outputSamplingGridTypeToUse as GridType
+    }
+
+    // filter [status, rasterResolution, utmZoneAdjust, mgrsBandAdjust] after products gotten. 
+    // TODO: Implement these into this function once the filtering endpoint supports it.
+    if(startDate !== 'none') {
+        productVariablesObject.afterTimestamp = new Date(startDate.setSeconds(0,0)).toISOString().replace('Z','')
+    }
+    if(endDate !== 'none') {
+        productVariablesObject.beforeTimestamp = new Date(endDate.setSeconds(0,0)).toISOString().replace('Z','')
+    }
+    return productVariablesObject as UserProductQueryVariables
+}
+
+export const productPassesFilterCheck = (currentFilters: FilterParameters, status: string, rasterResolution: number, utmZoneAdjust?: number, mgrsBandAdjust?: number): boolean => {
+    let productPassesFilter = true
+
+    if (currentFilters.status.length > 0 && !currentFilters.status.includes(status as ProductState)) {
+        productPassesFilter = false
```

Comment:
Tiny nitpick - couldn't we just return false here instead of waiting to go down the rest of the checks? Applies to the rest of the `productPassesFilter = false` lines below

Reply:
Very true, thank you for catching that